packages/block-serialization-default-parser/src/index.js

```diff
@@ -3,6 +3,7 @@ let offset;
 let output;
 let stack;
 
+/* eslint-disable jsdoc/check-values */
```

Comment:
What was the warning that was happening when this wasn't here? Is it not fixed by the changes to the `@type` line?

Reply:
The explanation `added optimization to prevent backtracking on attribute parsing` after `@since` tag isn't valid. 

Message below:

```
Invalid JSDoc @since: "4.6.1 added optimization to prevent backtracking on attribute parsing".eslint(jsdoc/check-values)
```

Reply:
Do you have any thoughts on how we might make this valid, while still maintaining the goals of including the message? I'm not sure if JSDoc has any standard way to describe updates in specific versions. It could be that we simply include this in the summary text.

e.g.

```
 * In 4.6.1, this was updated to add an optimization to prevent backtracking on
 * attribute parsing.
```

(Then the `@since 4.6.1` line could be removed)

Reply:
How about this way?

```js
/** 
 *
 * @since 3.8.0
 * 
 * # History
 * - 4.6.1 added optimization to prevent backtracking on attribute parsing
 */
```

Reply:
In lieu of any official JSDoc way of doing this, your suggestion seems reasonable to me.

A couple nit-picking points:

- As far as placement, it should probably be attached as part of the extended description. It might even be necessary to do so, to ensure that it's picked up by the parsing tools.
- We should test to see if the Markdown syntax works with how [docgen](https://github.com/WordPress/gutenberg/tree/master/packages/docgen) generates documentation, and if it does, whether we want it to be H1, since it's likely not the best heading level in the context of document. Depending if we want to _assume_ how the documentation is structured, it could either be H3 (`###`) or non-structured (bolded `**`).

>If we decide to use my proposal, then should we update the coding standard, too, for this?

It would be worth considering I think, yeah. And as such, it would be a good topic of a [future JavaScript meeting](https://make.wordpress.org/meetings/). I can see about adding it to the agenda for next week's meeting.

I was curious how this is managed in the WordPress PHP code. It appears the standard there is to add description as part of `@since`, just like what we're proposing to replace:

https://github.com/WordPress/wordpress-develop/blob/a44943546ca6259279f2632b1b4890bd7cc5ca22/src/wp-comments-post.php#L48-L49

Looking at the PHPDoc standard documentation for `@since`, I see this:

https://manual.phpdoc.org/HTMLSmartyConverter/HandS/phpDocumentor/tutorial_tags.since.pkg.html

What stands out to me is "[unspecified format]". It seems like something where the lack of a format imposed gives some potential flexibility by the project to decide to implement their own conventions.

But to come full circle, couldn't we do that as well? Looking at the JSDoc standard:

https://jsdoc.app/tags-since.html

There's not really an imposed restriction on what `<versionDescription>` could mean. You might even take "description" to mean that we _can_ include a summary of the changes. Maybe `eslint-plugin-jsdoc` is too strict, or it's something where we just want to opt out of this specific validation, because we're okay to include the summaries in the `@since` tag.

The more I think about it, the more I wonder if the simplest way to go about this is to just opt out of [`check-values`](https://github.com/gajus/eslint-plugin-jsdoc#check-values). What do you think?

Reply:
I was confused with `check-value` and `check-type`. I thought we need both of them to check types. But after reading `check-value` doc, `check-value` is for `@version`, `@since`, `@author`, `@license` tags. 

We don't use `@author`, `@license`, `@version` tags. So, what `check-value` option only does for Gutenberg project is to ensure if the version number follows the semver format. We can check this manually. I think it doesn't harm to opt out of `check-values`.

The best scenario might be `check-value` has an option like 'allowVersionChanges` and checks the version format automatically. But it doesn't exist now. 

As for this, we need to change these 2 things [in the doc](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/):

**From:**

> @since x.x.x: Should always be 3-digit (e.g. @since 3.6.0).

**To:**

```
@since x.x.x: Should always be 3-digit (e.g. @since 3.6.0). You can optionally add what is changed after the number (e.g. @since 11.0.0 Check server state by default.)
```

To let new developers know that they can leave changes as comments if it's important. 

And add one more `@since`:

```js
/**
 * @since      x.x.x
 * @since      x.x.x Check server state by default
 */
```

Because some people don't read explanation above and go straight to the example.

Reply:
> As for this, we need to change these 2 things [in the doc](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/):
> 
> **From:**
> 
> > @SInCE x.x.x: Should always be 3-digit (e.g. @SInCE 3.6.0).
> 
> **To:**
> 
> ```
> @since x.x.x: Should always be 3-digit (e.g. @since 3.6.0). You can optionally add what is changed after the number (e.g. @since 11.0.0 Check server state by default.)
> ```
> 
> To let new developers know that they can leave changes as comments if it's important.
> 
> And add one more `@since`:
> 
> ```js
> /**
>  * @since      x.x.x
>  * @since      x.x.x Check server state by default
>  */
> ```
> 
> Because some people don't read explanation above and go straight to the example.

To me, that sounds like a very sensible change to the standard 👍 

I'll plan to add it to the agenda for next week's JavaScript meeting.

In the meantime, would it be a good course of action here then to opt out of `check-values`?

I think in a perfect world, we could validate our `@version` standard (either `x.x.x` or `x.x.x` plus a specially formatted message). I imagine would likely require building something custom, so not something I think needs to be done here.

Reply:
I note there's a section about this tag in the PHP documentation standards:

https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#since-section-changelogs

Reply:
* I'll opt it out tomorrow. 
* I agree. Validating version/since should be a separate issue.

Reply:
To follow up, this topic was discussed during today's JavaScript meeting in the development Slack:

https://make.wordpress.org/core/2020/03/17/javascript-chat-summary-march-17-2020/

It was well-received. I don't anticipate there being any problems to update the documentation standards, and I'll plan to do so later in the week, assuming there is no opposition raised in the comments of the summary above.

Aside: You're welcomed to join our weekly chats if you'd like! They're held each Tuesday in the Slack instance ([meeting calendar](https://make.wordpress.org/meetings/), [chat registration](https://make.wordpress.org/chat/)).